### PR TITLE
Upgrade dependencies and add tests

### PR DESCRIPTION
**Pull request by shubham1556**

### CODE REVIEW

**Review by HimanshuGCvent — APPROVED** [review]

File: .tool-versions
Comment (MandarinConLaBarba) [inline .tool-versions:1]: Let's take this out

File: .gitignore
Comment (MandarinConLaBarba) [inline .gitignore:1]: Let's take this out

File: pnpm-lock.yaml
Comment (MandarinConLaBarba) [inline pnpm-lock.yaml:1]: Let's take this out

**Review by MandarinConLaBarba — CHANGES_REQUESTED** [review]
See inline comments

**Review by MandarinConLaBarba — APPROVED** [review]

**Pull request merged**